staging/src/k8s.io/apimachinery/pkg/runtime/serializer/protobuf/protobuf.go

```diff
@@ -176,7 +193,8 @@ func (s *Serializer) doEncode(obj runtime.Object, w io.Writer) error {
 	switch t := obj.(type) {
 	case *runtime.Unknown:
 		estimatedSize := prefixSize + uint64(t.Size())
-		data := make([]byte, estimatedSize)
+		data := getSlice(int(estimatedSize))
+		defer pool.Put(data)
```

Comment:
This isn't the right thing to do, the 2nd switch statement should attempt to reuse this buffer rather than make two trips to the pools.